includes/class-sensei-lesson.php

```diff
@@ -726,6 +729,141 @@ public function add_lesson_to_course_order( $lesson_id = 0 ) {
+	 * @return string The video thumbnail URL.
+	 */
+	private function get_featured_video_media_from_classic_editor( $url ) {
+		$url_parse = wp_parse_url( $url );
+
+		if ( false !== strpos( $url_parse['host'], 'youtube' ) ) {
+			return $this->get_youtube_thumbnail( $url );
+		}
+		if ( false !== strpos( $url_parse['host'], 'vimeo' ) ) {
+			return $this->get_vimeo_thumbnail( $url );
+		}
+		if ( false !== strpos( $url_parse['host'], 'videopress' ) ) {
+			return $this->get_videopress_thumbnail( $url );
+		}
+	}
+	/**
+	 * Get featured video url from the Featured Video Block
+	 *
+	 * @param int $post_id The post id.
+	 * @return string|null The URL string or null if the post does not have one.
+	 */
+	private function get_featured_video_media_from_blocks( $post_id ) {
+		$post   = get_post( $post_id );
+		$blocks = parse_blocks( $post->post_content );
+		foreach ( $blocks as $block ) {
+			if ( 'sensei-lms/featured-video' === $block['blockName'] ) {
+				if ( 'sensei-pro/interactive-video' === $block['innerBlocks'][0]['blockName'] ) {
+					$block = $block['innerBlocks'][0];
+				}
+				if ( 'core/video' === $block['innerBlocks'][0]['blockName'] ) {
+					if ( $block['innerBlocks'][0]['attrs']['videoPressClassNames'] ) {
+						return $block['attrs']['poster'];
+					} else {
+						return wp_get_attachment_url( get_post_thumbnail_id( $block['innerBlocks'][0]['attrs']['id'] ) );
+					}
+				}
+				if ( 'core/embed' === $block['innerBlocks'][0]['blockName'] ) {
+					$url = $block['innerBlocks'][0]['attrs']['url'];
+					if ( 'youtube' === $block['innerBlocks'][0]['attrs']['providerNameSlug'] ) {
+						return $this->get_youtube_thumbnail( $url );
+					} elseif ( 'vimeo' === $block['innerBlocks'][0]['attrs']['providerNameSlug'] ) {
+						return $this->get_vimeo_thumbnail( $url );
+					} elseif ( 'videopress' === $block['innerBlocks'][0]['attrs']['providerNameSlug'] ) {
+						return $this->get_videopress_thumbnail( $url );
+					}
+				}
+			}
+			return null;
+		}
+	}
+
+	/**
+	 * Save Lesson Featured Video thumbnail to post meta
+	 *
+	 * @param int $post_id The Post Id.
+	 */
+	public function save_lesson_featured_video_thumbnail( $post_id ) {
+		$meta_key       = '_featured_video_thumbnail';
+		$thumbnail_meta = get_post_meta( $post_id, $meta_key, true );
+
+		if ( has_blocks( $post_id ) ) {
+			$thumbnail = $this->get_featured_video_media_from_blocks( $post_id );
+		} else {
+			$video_embed = get_post_meta( $post_id, '_lesson_video_embed', true );
+			if ( $video_embed ) {
+				$thumbnail = $this->get_featured_video_media_from_classic_editor( $video_embed );
+			}
+		}
+		if ( null !== $thumbnail && $thumbnail !== $thumbnail_meta ) {
+			update_post_meta( $post_id, $meta_key, $thumbnail );
+		}
+	}
+
+	/**
+	 * Get the featured video thumbnail URL from a Post's metadata.
+	 *
+	 * @param int $post_id The Post ID.
+	 * @return string The video thumbnail URL.
+	 */
+	public static function get_featured_video_thumbnail_url( $post_id ) {
```

Comment:
Is this function better off in `Sensei_Utils` class since it's static and not necessarily tied to any specific lesson?

Reply:
`Sensei_Utils` file is ~2600 lines of code now. I don't think it's wise to keep adding methods into it.

If you move this method exactly the way it is from `Sensei_Utils` to `Lesson_Video` class, it will still be not tied to any specific lesson.

If it doesn't feel right to put this method in `Lesson_Video` then we could introduce a class called `Sensei_Lesson_Utils` or something. I would not prefer that. I think the `utils` or `helpers` keywords should be used sparingly, because they are too broad and basically anything could be considered as a `util` function and end up in there.

Reply:
Good call, @onubrooks - I'll make the change! :)